openedx/adg/lms/applications/api/serializers.py

```diff
@@ -8,7 +8,32 @@
 from openedx.adg.lms.applications.models import Education, Reference, WorkExperience
 
 
-class EducationSerializer(serializers.ModelSerializer):
+class ApplicationRequirementsSerializer(serializers.ModelSerializer):
+    """
+    Base serializer to add `user_application` while validating the data for Education, Work Experience and References
+    while creating and updating objects.
+    """
+
+    def validate(self, attrs):
+        """
+        Adds `user_application` to the data using the `request.user`.
+
+        Args:
+            attrs (dict): Dict containing data to be validated
+
+        Returns:
+            dict: Validated data
+        """
+        user = self.context['request'].user
+
+        if hasattr(user, 'application'):
+            attrs['user_application'] = user.application
+        else:
+            raise serializers.ValidationError('User has no written application')
```

Comment:
For future reference, a good practice to add a blank line before return statement for better readability.